addons/product/models/product_pricelist.py

```diff
@@ -345,6 +345,15 @@ def get_import_templates(self):
             'label': _('Import Template for Pricelists'),
             'template': '/product/static/xls/product_pricelist.xls'
         }]
+    
+    def unlink(self):
+        for pricelist in self:
+            linked_items = self.env['product.pricelist.item'].sudo().with_context(active_test=False).search(
+                [('base', '=', 'pricelist'), ('pricelist_id', '=', pricelist.id)])
```

Comment:
```suggestion
                [('base', '=', 'pricelist'), ('base_pricelist_id', '=', pricelist.id)])
```
Your current domain forbids the deletion of a pricelist if one of its rules is based on another pricelist when it should forbid the deletion of a pricelist if the rule of another pricelist is based on it (Only noticed now, my bad :man_facepalming:)